### PR TITLE
Close #979 by adding encoders and decoders for MonthDay, Year, and ZoneOffset

### LINKED ISSUE #979: Instances for java.time.MonthDay
Seems to be missing: 
https://docs.oracle.com/javase/8/docs/api/java/time/MonthDay.html

### PR DESCRIPTION
**Pull request by plokhotnyuk**
...and improve throughput of parsing and serialization of YearMonth values, by using build-in parser and formatter

### CODE REVIEW

**Comment by travisbrown** [comment]
Thanks @plokhotnyuk! Will get this merged shortly.

File: modules/java8/src/main/scala/io/circe/java8/time/JavaTimeDecoders.scala [hunk 1]
```diff
@@ -124,6 +127,15 @@ private[time] trait JavaTimeDecoders {
         LocalDateTime.parse(input, formatter)
     }
 
+  /**
+   * @group Time
+   */
+  final def decodeMonthDayTimeWithFormatter(formatter: DateTimeFormatter): Decoder[MonthDay] =
```
Comment (travisbrown) [inline modules/java8/src/main/scala/io/circe/java8/time/JavaTimeDecoders.scala]: Sorry about the nitpick, but do you mind changing this to `decodeMonthDayWithFormatter` for consistency? (I can also do it in a follow-up.)

Comment (plokhotnyuk) [reply modules/java8/src/main/scala/io/circe/java8/time/JavaTimeDecoders.scala]: Nice spot! I will fix it now.

**Comment by plokhotnyuk** [comment]
Also, I fixed formatting of comments and squashed commits.

As W/A it would better to add more tests for Java Time API codecs with custom formatters.

**Comment by travisbrown** [comment]
> As W/A it would better to add more tests for Java Time API codecs with custom formatters.

Agreed.

**Pull request merged**